assets/source/tests/tracking.test.js

```diff
@@ -0,0 +1,47 @@
+jest.mock( '../setup-guide/app/helpers/effects', () => {
+	return {
+		useCreateNotice: () => () => {},
+		useSettingsSelect: () => {},
+		useSettingsDispatch: () => () => {},
+	};
+} );
+jest.mock( '@woocommerce/tracks', () => {
+	return {
+		recordEvent: jest.fn(),
+	};
+} );
+
+jest.mock( '@wordpress/api-fetch', () => {
+	return {
+		default: jest.fn(),
+	};
+} );
+
+/**
+ * External dependencies
+ */
+import { recordEvent } from '@woocommerce/tracks';
+import apiFetch from '@wordpress/api-fetch';
+import { fireEvent, render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import ClaimWebsite from '../setup-guide/app/steps/ClaimWebsite';
+
+describe( 'Claim Website Record Events', () => {
+	it( 'pfw_domain_verify_failure is called on domain verification failure', () => {
+		apiFetch.default.mockImplementation( async () => {
+			throw 'Ups';
+		} );
+
+		const { getByText } = render(
+			<ClaimWebsite goToNextStep={ () => {} } view="wizard" />
+		);
+
+		fireEvent.click( getByText( 'Start Verification' ) );
+		expect( recordEvent.mock.calls[ 0 ][ 0 ] ).toBe(
+			'pfw_domain_verify_failure'
+		);
+	} );
+} );
```

Comment:
This is because in 

https://github.com/woocommerce/pinterest-for-woocommerce/pull/299/files#diff-48213ce8e813be7f415db4778226cf84503c4ae235f961201e8ca8d68195b803

I realised that I was duplicating several Mocks and logic, so I thought a good way would be to have that tracking file testing everything. Let me know your thoughts.

Reply:
Personally, I think it would be easier to maintain unit tests in files that are associated directly with the React component they refer to, rather than have it organized horizontally by features.
If we have duplicated mocks we could move that to helpers to be shared between individual React component's tests.

What logic do you have in mind precisely?

```
jest.mock( '@woocommerce/tracks', () => {
	return {
		recordEvent: jest.fn(),
	};
} );
```
can be reduced to a single liner: `jest.mock( '@woocommerce/tracks' )`.

and 
```js
jest.mock( '@woocommerce/settings', () => {
	return {
		getSetting: () => {
			return {
				es: 'Spain',
			};
		},
	};
} );
```

Could be just 
```js
jest.mock( '@woocommerce/settings' );
// …
getSetting.mockReturnValue({ es: 'Spain' } );
```
Or we can make `assets/source/tests/dependencies/woocommerce/settings.js` return a jest mock, so we will not need to mock it here at all.

Reply:
Understood. WDYT of merging this and #299 and then after that I can do the refactor? Since #299 also uses `tracking.test.js `and would be easier for me that way.

I can do that today :)  

Created issue https://github.com/woocommerce/pinterest-for-woocommerce/issues/306 for that

Reply:
I like the idea :)